jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/utility/BluetoothDateTime.kt

```diff
@@ -0,0 +1,34 @@
+package org.ehealthinnovation.jdrfandroidbleparser.bgm.utility
+
+import java.util.*
+
+class BluetoothDateTime {
+    companion object {
+        /**
+         * Convert bluetooth date time fields into Java Date format. It takes care of the offset addition.
+         * @param y Year in bluetooth definition
+         * @param m Month in bluetooth definition
+         * @param d Day in bluetooth definition
+         * @param hr Hour in bluetooth definition
+         * @param min Minute in bluetooth definition
+         * @param sec Second in blutooth definition
+         * @param offset the offset
+         * @return the date of the measurement
+         * @see https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.date_time.xml
+         */
+        fun getDate(y:Int?, m:Int?, d:Int?, hr:Int?, min:Int?, sec:Int?, offset:Int?): Date? {
+            if(y==null || m==null || d==null || hr==null || min==null || sec==null)
+                return null
```

Comment:
Take the ? out of the constructor. then you don't have to do the null check.

Reply:
Again with this one, it's not a good method. It's an anti-pattern.

you also have convert to date as well as get date? Why?

Reply:
I have removed this method. It is no longer used. Good catch.